### PR TITLE
feat(addons): add file names, line, and col to keyAlreadyExists err

### PR DESCRIPTION
**Pull request by efekarakus**
When two addons templates define the same resource but with different values, the error now displays the file names and line and column where the mismatch happened. 

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by efekarakus** [comment]
Example with the new error:
```bash
✘ retrieve addons template: resource "MyTable" defined in "ddb.yaml" at Ln 34, Col 9 is different than in "s3.yaml" at Ln 3, Col 9
```

File: internal/pkg/addons/cloudformation.go [hunk 1]
```diff
@@ -103,19 +104,39 @@ func (t *cfnTemplate) mergeTransform(transform yaml.Node) error {
...
+			return
+		}
+		t.templateNameFor[node] = tplName
+		for _, c := range node.Content {
+			assign(c)
+		}
 	}
-	return nil
+
+	assign(&t.Metadata)
```
Comment (iamhopaul123) [inline internal/pkg/addons/cloudformation.go:133]: I was thinking maybe we can simplify by:

1) starting from the entry node for this template and do the bst (assuming we don't think it is valid to have "nodeA" defined in both `Metadata` of templateA and `Parameters` of templateB) 

or

2) make it a new field like `t.Content map[string]*node` instead to have `t.Parameters`, `t.Mappings` etc. By doing this we can change it to two-line for loop for this `assignNewNodesTo` and also we can remove `mergeMetadata` etc. to have a generic function with handler as a param.

What do you think of it 🤔

Comment (efekarakus) [reply internal/pkg/addons/cloudformation.go]: I used reflection to call `assign` on all the fields that represent CFN sections. Take a look at it and let me know what you think.

This way we don't have to update this function if we have to add new section

Comment (iamhopaul123) [reply internal/pkg/addons/cloudformation.go]: Makes sense to me :+1: Thank you for the extra efforts to make this change!

**Pull request merged**